### ISSUE TITLE
$(seq ...) quoting checks should be relaxed

### ISSUE THREAD

**Issue by vapier** [koalaman/shellcheck#696]
a common idiom is to use the output of `$(seq ...)` as a list of numbers.  shellcheck is unhappy with this though and wants it to be quoted which pretty much never makes sense.

```sh
$ cat test.sh
#!/bin/sh
printf '%s\n' $(seq 1 20)

$ shellcheck test.sh 
In test.sh line 2:
printf '%s\n' $(seq 1 20)
              ^-- SC2046: Quote this to prevent word splitting.
```

**Comment by rawiriblundell** [comment]
You'll note the examples given in the linked wiki page use double quotes.  I almost always use double quotes with `printf` myself, and this works perfectly fine:

```
$ printf "%s\n" "$(seq 1 20)"
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
```

I've also had a quick google-sanity-check for `printf` basics and they all seem to favour double quotes.  Single quoting will get you a literal printout of `seq 1 20`, as it should.

**Comment by vapier** [comment]
i think you're confused.  the quotes on the printf string are irrelevant.  ignoring that, you're misreading the output.  try this instead and you'll see that quoting the seq is wrong:
```sh
$ printf '{%s}\n' "$(seq 1 5)"
{1
2
3
4
5}
```

**Comment by kurahaupo** [comment]
On Tue, 21 Jun 2016, Mike Frysinger wrote:
> a common idiom is to use the output of `$(seq ...)` as a list of numbers.  shellcheck is unhappy with this though and wants it to be quoted which pretty much never makes sense.

Indeed; the more appropriate response would be to warn against using
```seq``` at all, especially if its range is not a constant.

Of course, that begs the question of what to replace it with. In Bash one
can use:
```
    for (( i=0 ; i<10 ; ++i )) do ...
```

In other posix shells one can use
```
    i=-1
    while
      i=$(($i+1))
      [ $i -lt 100 ]
    do ...
```

**Comment by vapier** [comment]
in some specific cases, using `for`/`while` loops are doable, but in many cases they're a clumsy replacement for `$(seq ...)`, especially when it's being passed directly to other commands (e.g. the `printf` example which isn't all that naive when you want formatted values).

the closest you'd get is brace expansion like `{0..10}`, but that doesn't play well with variables (`{0..$limit}` which requires `eval` which is even worse), and the increment field is new to bash 4.0 (`{0..10..2}` like `seq 0 2 10`).  and for people writing POSIX shell, it doesn't work at all :(.

**Comment by koalaman** [comment]
Obviously `seq` isn't POSIX either, but I agree that this case should not be giving a warning.

**Comment by vapier** [comment]
fair point that `seq` isn't defined by POSIX.  i think its lineage is sufficient at this point though for "portable" scripts to utilize :).  it first showed up in [1985](https://en.wikipedia.org/wiki/Seq_(Unix)) and has made its way into all existing BSD/Linux systems.

**Comment by koalaman** [comment]
Fixed in 50084c0
